### PR TITLE
Add date input for timeline event dates

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/20289

Changes:
- Adds `DateInput`. Borrowed some date formatting logic from `SpecificDatesPicker`.
- Adds `date` form field
- Uses it for timeline events

How to test:
- Go to a collection. Click on the calendar icon
- If there are no timelines, click `Add an event`. Otherwise you need to navigate to a timeline first.
- Make sure it's possible to enter the date with keyboard
- Make sure it's possible to select a date with a calendar (mouse only, not accessible)

Follow ups:
- Add time & timezone selection

Screenshot 2022-02-16 at 20 51 54

### CODE REVIEW

File: frontend/src/metabase/core/components/Input/Input.tsx [hunk 1]
```diff
@@ -7,37 +7,32 @@ export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
...
   return (
-    <InputRoot className={className} fullWidth={fullWidth}>
+    <InputRoot ref={ref} className={className} fullWidth={fullWidth}>
       <InputField
         {...rest}
         hasError={error}
         hasTooltip={Boolean(helperText)}
         fullWidth={fullWidth}
-        ref={ref}
+        borderless={borderless}
```
Comment (kulyk) [inline frontend/src/metabase/core/components/Input/Input.tsx:25]: If the `ref` is moved from `InputField` to `InputRoot`, won't it break a possibility to trigger `focus()` and `blur()` programmatically?

**Review by kulyk — COMMENTED** [review]
Nice! A few small things noticed:

1. Can we make sure we respect instance localization settings (`/admin/settings/localization`)? E.g. date separators 
2. It's really cool I can both pick a date with a calendar component or with a keyboard. It seems like the value set with the keyboard is not synced with the calendar though, please take a look:

https://user-images.githubusercontent.com/17258145/154458826-538a9738-9927-4a94-becf-de726a701e26.mp4

File: frontend/src/metabase/core/components/Input/Input.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/core/components/Input/Input.tsx:25]: Yes I believe so. If someone needs this, we should add another ref, e.g. `inputRef` as a prop. It' incorrect to forward refs to a nested element.

**Comment by ranquild** [comment]
@kulyk Fixed the issue with date selection, please have a look

File: frontend/src/metabase/core/components/Input/Input.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/core/components/Input/Input.tsx:25]: Do you think we should add `inputRef` in this PR?

**Review by kulyk — APPROVED** [review]
Looks good!

File: frontend/src/metabase/core/components/Input/Input.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/core/components/Input/Input.tsx]: Added

**Pull request merged**